core/lib/generators/solidus/install/install_generator.rb

```diff
@@ -187,34 +172,37 @@ def create_database
     end
 
     def install_frontend
-      bundler_context = BundlerContext.new
-
       if options[:frontend] == 'none'
-        support_solidus_frontend_extraction(bundler_context)
+        support_solidus_frontend_extraction
       else
-        frontend = detect_frontend_to_install(bundler_context)
+        frontend = detect_frontend_to_install
 
-        support_solidus_frontend_extraction(bundler_context) unless frontend == LEGACY_FRONTEND
+        support_solidus_frontend_extraction unless frontend == LEGACY_FRONTEND
 
         say_status :installing, frontend
 
         InstallFrontend
           .new(bundler_context: bundler_context, generator_context: self)
           .call(frontend)
+
+        # The DEFAULT_FRONTEND installation makes changes to the
+        # bundle without updating the bundler context. As such, we need to
+        # reset the bundler context to get the latest dependencies from the
+        # context.
+        reset_bundler_context if frontend == DEFAULT_FRONTEND
```

Comment:
From the comment:

```
        # The DEFAULT_FRONTEND installation makes changes to the
        # bundle without updating the bundler context. As such, we need to
        # reset the bundler context to get the latest dependencies from the
        # context.
```

Reply:
@waiting-for-dev It's needed because BundlerContext stores the dependencies in an instance variable. If we want to get the latest dependencies, we'd have to create a new BundlerContext.

One alternative is to change `BundlerContext#dependencies` so that it always returns the latest dependencies:

```ruby
class BundlerContext
  def dependencies
    Bundler.definition.locked_gems.dependencies
  end 
end
```

However, I'm guessing BundlerContext stores dependencies in an instance variable for performance reasons i.e. caching.

Your thoughts?

Reply:
Ah, I see. Then, what isn't needed is the change in https://github.com/solidusio/solidus/pull/4748/files#r1034556084, right? (I'd still memorize and invalidate as it probably makes sense for performance reasons)

Reply:
We still need https://github.com/solidusio/solidus/pull/4748/files#r1034556084 (changing `Bundler.locked_gems` to `Bundler.definition.locked_gems`) since `Bundler.locked_gems` also memoizes the locked gems (see https://rubydoc.info/gems/bundler/2.3.22/Bundler.locked_gems). If we kept `Bundler.locked_gems`, a new instance of BundlerContext would load the memoized locked gems from Bundler, with its corresponding dependencies.

Reply:
Got it. Thanks for the explanation, @gsmendoza.